google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/RestoreTableRequest.java

```diff
@@ -20,24 +20,47 @@
 import com.google.common.base.Objects;
 import com.google.common.base.Preconditions;
 import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 
 /** Fluent wrapper for {@link com.google.bigtable.admin.v2.RestoreTableRequest} */
 public final class RestoreTableRequest {
   private final com.google.bigtable.admin.v2.RestoreTableRequest.Builder requestBuilder =
       com.google.bigtable.admin.v2.RestoreTableRequest.newBuilder();
-  private final String backupId;
-  private final String clusterId;
+  private final String sourceBackupId;
+  private final String sourceClusterId;
+  private final String sourceInstanceId;
 
-  public static RestoreTableRequest of(String clusterId, String backupId) {
-    RestoreTableRequest request = new RestoreTableRequest(clusterId, backupId);
+  /**
+   * Create a {@link RestoreTableRequest} object. It assumes the source backup locates in the same
+   * instance as the destination table. To restore a table from a backup in another instance, use
+   * {@link #of(String, String, String) of} method.
+   */
+  public static RestoreTableRequest of(String sourceClusterId, String sourceBackupId) {
+    RestoreTableRequest request = new RestoreTableRequest(null, sourceClusterId, sourceBackupId);
     return request;
   }
 
-  private RestoreTableRequest(String clusterId, String backupId) {
-    Preconditions.checkNotNull(clusterId);
-    Preconditions.checkNotNull(backupId);
-    this.backupId = backupId;
-    this.clusterId = clusterId;
+  /**
+   * Create a {@link RestoreTableRequest} object. It assumes the source backup locates in the same
+   * instance as the destination table. To restore a table from a backup in another instance, use
+   * {@link #of(String, String, String) of} method.
+   */
+  public static RestoreTableRequest of(
+      String sourceInstanceId, String sourceClusterId, String sourceBackupId) {
```

Comment:
what are your thoughts on renaming `sourceInstanceId` to `destinationInstanceId`, or `targetInstanceId`? 'source' is a little confusing to me

Reply:
ignore - I misunderstood the usage here